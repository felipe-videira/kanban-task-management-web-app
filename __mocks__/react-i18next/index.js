export const useTranslation = () => ({ t: (str) => str });
