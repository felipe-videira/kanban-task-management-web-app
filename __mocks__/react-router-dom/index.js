export const navigateMock = jest.fn();
export const useParamsMock = jest.fn();

export const useParams = useParamsMock;
export const useNavigate = () => navigateMock;
