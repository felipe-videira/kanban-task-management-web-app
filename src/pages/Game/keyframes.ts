import { keyframes } from "styled-components/macro";

export const fadeIn = keyframes`
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }`;

export const stepFadeIn = keyframes`
  0% {
    display: none;
    opacity: 0;
  }
  50% {
    display: block;
  }
  100% {
    opacity: 1;
  }
`;

export const growAndfadeIn = keyframes`
  0%  {
   	width: 0;
  	height: 0;
  	opacity: 0;
  }
  25% {
   width: 100px;
   height: 100px;
   opacity: 0;
  }
  100% {
  	opacity: 1;
  }
`;

export const radialBackgroundEffect = keyframes`
   0% {
      background: radial-gradient(circle, rgba(255,255,255,0.2) 0%, rgba(255,255,255,0.2) 37.5%, rgba(255,255,255,0.15) 38%, rgba(255,255,255,0) 38%);
      opacity: 0;
  }
  25% {
    background: radial-gradient(circle, rgba(255,255,255,0.2) 0%, rgba(255,255,255,0.2) 37.5%, rgba(255,255,255,0.15) 38%, rgba(255,255,255,0.05) 55%, rgba(255,255,255,0.025) 56%,  rgba(255,255,255,0) 56%);
    opacity: 0.25;
  }
 50% {
    background: radial-gradient(circle, rgba(255,255,255,0.2) 0%, rgba(255,255,255,0.2) 37.5%, rgba(255,255,255,0.15) 38%, rgba(255,255,255,0.1) 55%, rgba(255,255,255,0.05) 56%,  rgba(255,255,255,0) 56%);
    opacity: 0.5;
  }
   75% {
     background: radial-gradient(circle, rgba(255,255,255,0.2) 0%, rgba(255,255,255,0.2) 37.5%, rgba(255,255,255,0.15) 38%, rgba(255,255,255,0.1) 55%, rgba(255,255,255,0.025) 56%, rgba(255,255,255,0.012.5) 75%, rgba(255,255,255,0.0) 76%);
     opacity: 0.75;
  }
  100% {
     background: radial-gradient(circle, rgba(255,255,255,0.2) 0%, rgba(255,255,255,0.2) 37.5%, rgba(255,255,255,0.15) 38%, rgba(255,255,255,0.1) 55%, rgba(255,255,255,0.05) 56%, rgba(255,255,255,0.025) 75%, rgba(255,255,255,0.0) 76%);
     opacity: 1;
  }
`;
