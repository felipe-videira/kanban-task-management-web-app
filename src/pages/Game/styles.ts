import styled, { css } from "styled-components/macro";
import { mobile } from "../../utils/breakpoints";
import {
  fadeIn,
  stepFadeIn,
  growAndfadeIn,
  radialBackgroundEffect,
} from "./keyframes";

//TODO: separar
export const Container = styled.div`
  animation: ${fadeIn} 2s linear 0s 1;
  animation-fill-mode: both;

  display: flex;
  align-items: center;
  justify-content: space-evenly;
  flex-direction: column;
  overflow: hidden;
  margin: 0 auto;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 100vh;
  padding: 0 5%;

  ${mobile} {
    padding: 0;
  }
}
`;

export const Display = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  border: 3px solid #ffffff59;
  border-radius: 20px;
  padding: 2.5% 5%;

  ${mobile} {
    width: 75%;
  }
`;

export const Title = styled.h1`
  text-transform: uppercase;
  width: 10%;
`;

interface ScoreProps {
  readonly value: number;
}

export const Score = styled.div<ScoreProps>`
  &::after {
    content: "${(props) => props.value}";
  }
`;

interface OptionsProps {
  readonly size: number;
}

export const Options = styled.div<OptionsProps>`
  ${(props) => `
    height: ${props.size}px;
    width: ${props.size}px;
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 100px;
`}
`;

interface StepperProps {
  readonly value: number;
}

export const Step = styled.div<StepperProps>`
  animation: ${stepFadeIn} 1s linear 0s 1;
  animation-fill-mode: both;
  display: none;
`;

export const Stepper = styled.div<StepperProps>`
  ${(props) => `
    ${Step}[value="${props.value}"] {
      display: block;
    }
`}
`;

const GameResultChoice = css`
  width: 100px;
  height: 100px;
  background-color: red;
  border-radius: 50%;
`;

const GameResulChoicetWinner = css`
  position: relative;

  &::after {
    content: "";
    width: 300%;
    height: 300%;
    background: radial-gradient(
      circle,
      rgba(255, 255, 255, 0.2) 0%,
      rgba(255, 255, 255, 0.2) 37.5%,
      rgba(255, 255, 255, 0.15) 38%,
      rgba(255, 255, 255, 0.1) 55%,
      rgba(255, 255, 255, 0.05) 56%,
      rgba(255, 255, 255, 0.025) 75%,
      rgba(255, 255, 255, 0) 76%
    );
    position: absolute;
    border-radius: 50%;
    top: -100%;
    left: -100%;
    z-index: -1;
    animation: ${radialBackgroundEffect} 2s linear 3s infinite alternate;
    animation-fill-mode: both;
  }
`;

export const GameResultUserChoice = styled.div`
  ${GameResultChoice}
`;

export const GameResultHouseChoice = styled.div`
  ${GameResultChoice}
  animation-name: ${fadeIn};
  animation-fill-mode: both;
`;

export const GameResult = styled.div`
  align-items: center;
  justify-content: center;
  display: flex;
  width: 100px;
  height: 100px;
  animation-name: ${growAndfadeIn};
  animation-fill-mode: both;
  z-index: 1;
`;

interface GameResultProps {
  readonly userWins: boolean;
  readonly delayInSecs: number;
}

export const GameResultContainer = styled.div<GameResultProps>`
  display: flex;

  ${GameResultHouseChoice} {
    animation-duration: ${(props) => props.delayInSecs / 2}s;
    animation-delay: ${(props) => props.delayInSecs / 4}s;
  }

  ${GameResult} {
    animation-delay: ${(props) => props.delayInSecs / 2}s;
    animation-duration: ${(props) => props.delayInSecs / 4}s;
  }

  ${(props) =>
    props.userWins ? GameResultUserChoice : GameResultHouseChoice} {
    ${GameResulChoicetWinner}
  }
`;
