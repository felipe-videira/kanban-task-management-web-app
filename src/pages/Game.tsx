import { useMemo } from "react";
import { useParams } from "react-router-dom"
import gameConfig from '../gameConfig.json';

function Game() {
    const { gameName } = useParams();

    const game = useMemo(() => gameConfig.find(config => config.name === gameName), []);

    return (
      <div>
          {game?.options.map(option => (
            <div>{option.name}</div> 
          ))}
      </div>
    )
  }
  
  export default Game
  