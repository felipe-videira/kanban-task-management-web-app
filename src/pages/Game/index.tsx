import { useEffect, useState } from "react";
import { useParams, useNavigate } from "react-router-dom";
import Option from "../../components/Option";
import PolygonalList from "../../components/PolygonalList";
import gameConfig from "../../gameConfig.json";
import * as userScoreService from "../../services/userScore";
import { clamp } from "../../utils/clamp";
import { isMobile } from "../../utils/isMobile";
import { random } from "../../utils/random";
import {
  Container,
  Score,
  Options,
  Display,
  Title,
  Stepper,
  Step,
  GameResultContainer,
  GameResultUserChoice,
  GameResult,
  GameResultHouseChoice,
} from "./styles";

const RESULT_DELAY = 6;

type GameOption = {
  name: string;
  icon: string;
  color: string[] | string;
  winRules: {
    [key: string]: boolean;
  };
};

type GameConfig = {
  name: string;
  rules: string;
  listSetup: {
    pointingUp: boolean;
  };
  options: GameOption[];
};

function Game() {
  const navigate = useNavigate();
  const { gameName } = useParams();

  const [game, setGame] = useState<GameConfig>();
  const [listSize, setListSize] = useState<number>(0);
  const [optionSize, setOptionSize] = useState<number>(0);
  const [userScore, setUserScore] = useState<number>(0);
  const [userChoice, setUserChoice] = useState<string>();
  const [houseChoice, setHouseChoice] = useState<string>();
  const [userWins, setUserWins] = useState<boolean>(false);
  const [step, setStep] = useState<number>(1);

  function getSize(
    selectedGame: GameConfig,
    screenWidth: number = screen.width
  ): [optionsSize: number, listSize: number] {
    const { length } = selectedGame.options;
    const size = clamp(
      (screenWidth / length) * (isMobile() ? 1 : 0.4),
      50,
      isMobile() ? screenWidth / length : screenWidth / 3 / length
    );

    return [size, size * (length / 2)];
  }

  function playMatch(
    optionName: string,
    options: GameOption[]
  ): [isWinner: boolean, houseOption: GameOption] {
    const houseOption = options[random(0, options.length - 1)];
    const userOption = options.find((option) => option.name === optionName);
    return [!!userOption?.winRules[houseOption.name], houseOption];
  }

  //todo
  function incrementScore() {
    //todo
    if (userWins && game) {
      const newScore = userScore + 1;
      setUserScore(newScore);
      userScoreService.save(game.name, newScore);
    }
  }

  function onOptionClick(name: string) {
    const [isWinner, houseOption] = playMatch(name, game?.options || []);

    setUserChoice(name);
    setHouseChoice(houseOption.name);
    setUserWins(isWinner);
    setStep(2);
  }

  function resetGame() {
    setStep(1);
    setUserChoice(undefined);
    setHouseChoice(undefined);
    setUserWins(false);
  }

  function setupGame(): GameConfig | undefined {
    const selectedGame = gameConfig.find(
      (config) => config.name === gameName
    ) as GameConfig | undefined;

    if (selectedGame) {
      setGame(selectedGame);

      const [options, list] = getSize(selectedGame);
      setOptionSize(options);
      setListSize(list);
      setUserScore(userScoreService.get(selectedGame.name));
    }

    return selectedGame;
  }

  useEffect(() => {
    const selectedGame = setupGame();

    if (selectedGame) {
      userScoreService.setOnExitSaveListener(() => {
        userScoreService.save(selectedGame.name, userScore);
      });
    } else {
      navigate("/");
    }
  }, []);

  return (
    <Container>
      <Display>
        <Title>{game?.name}</Title>
        <Score value={userScore} />
      </Display>
      <Options size={listSize}>
        <Stepper value={step}>
          <Step value={1}>
            <PolygonalList
              data={game?.options || []}
              ItemComponent={Option}
              itemProps={{
                size: optionSize,
                onClick: onOptionClick,
              }}
              itemSize={optionSize}
              size={listSize}
              pointingUp={game?.listSetup.pointingUp}
            />
          </Step>
          <Step value={2}>
            <GameResultContainer userWins={userWins} delayInSecs={RESULT_DELAY}>
              <GameResultUserChoice>{userChoice}</GameResultUserChoice>
              <GameResult onAnimationEnd={incrementScore}>
                <p>{userWins ? "You win" : "You loose"}</p>
                <button type="button" onClick={resetGame}>
                  Play again
                </button>
              </GameResult>
              <GameResultHouseChoice>{houseChoice}</GameResultHouseChoice>
            </GameResultContainer>
          </Step>
        </Stepper>
      </Options>
    </Container>
  );
}

export default Game;
