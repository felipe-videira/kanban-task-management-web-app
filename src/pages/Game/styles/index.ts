import styled from "styled-components/macro";
import {
  growAndfadeIn,
  moveLeft,
  moveRight,
  radialBackgroundEffect,
} from "./keyframes";
import Button from "../../../components/Button";
import { LogoAltText, LogoImg } from "../../../components/Logo";
import { OptionButton } from "../../../components/Option/styles";
import _Stepper, { Step } from "../../../components/Stepper";
import { mobile, phone, phoneSm, tablet } from "../../../utils/breakpoints";
import { fadeIn } from "../../../utils/keyframes";

type ResultProps = {
  readonly userWins: boolean;
  readonly showHouseChoiceDelay: number;
  readonly showHouseChoiceDuration: number;
  readonly showResultDelay: number;
  readonly showResultDuration: number;
  readonly winnerBackgroundEffectDelay: number;
  readonly size: number;
};

export const Container = styled.div`
  animation: ${fadeIn} 0.2s linear 0s 1 normal both;
  margin: 0 auto;
  display: grid;
  align-items: center;
  justify-content: center;
  flex: 1;
  grid-template:
  ". header ."
  ". game rules";
  grid-template-columns: 1fr 60% 1fr;
  grid-template-rows: max-content 1fr;
  width: 100%;
  padding: 2rem 0 1rem;
  z-index: 0;
  
  ${mobile} {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }
}
`;

export const RulesButton = styled(Button).attrs(() => ({
  outlined: true,
  small: true,
}))`
  align-self: end;
  justify-self: end;
  grid-area: rules;
  margin: 0 10%;
  border-radius: 8px;
  min-width: 130px;
  width: auto;
  z-index: 2;

  ${mobile} {
    align-self: center;
    margin: 0;
  }.
`;

export const Header = styled.div`
  border: 3px solid ${(props) => props.theme.contrast.low};
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-radius: 15px;
  padding: 2%;
  grid-area: header;
  width: 70%;
  margin: 0 auto;
  z-index: 2;
  flex-wrap: wrap;
  gap: 1rem 0;

  ${LogoImg}, ${LogoAltText} {
    flex: 0.25;
  }

  ${phone} {
    border-radius: 7px;
    padding: 10px;
  }
`;

export const ScoreValue = styled.div`
  color: ${(props) => props.theme.dark};
  animation: ${fadeIn} 1s linear 0s 1 normal both;
  font-weight: bold;

  font-size: 4.5rem;
  line-height: 4.5rem;

  ${phone} {
    font-size: 2rem;
    line-height: 2rem;
  }

  ${phoneSm} {
    font-size: 1.5rem;
    line-height: 1.5rem;
  }

  &:first-child {
    order: 1;
  }

  &:last-child {
    order: 3;
  }
`;

export const Score = styled.div`
  background: ${(props) => props.theme.primary};
  min-width: 17.5%;
  border-radius: 7px;
  flex-wrap: wrap;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 2.5%;

  ${phone} {
    border-radius: 3px;
    max-width: 125px;
  }

  ${ScoreValue} + ${ScoreValue}::before {
    content: "x";
    font-size: 2rem;
    color: ${(props) => props.theme.dark};
    font-weight: bold;
    margin: 1rem;

    ${phone} {
      font-size: 1.5rem;
    }

    ${phoneSm} {
      font-size: 1.25rem;
    }
  }
`;

export const ScoreLabel = styled.h2`
  font-size: 1.2rem;
  color: ${(props) => props.theme.contrast.high};
  text-transform: uppercase;
  letter-spacing: 1.5px;
  width: 100%;
  text-align: center;
  margin: 0;
  margin-block: 0;
  font-weight: bold;

  ${phone} {
    font-size: 0.7rem;
  }

  ${phoneSm} {
    font-size: 0.5rem;
  }
`;

export const Stepper = styled(_Stepper)`
  grid-area: game;
  z-index: 1;
  flex: 1;
  align-self: stretch;
  display: flex;
  align-items: stretch;

  ${Step} {
    height: unset;
  }
`;

export const Options = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}
`;

export const RulesImageContainer = styled.div`
  svg {
    width: 100%;
  }
`;

const ResultChoiceLabel = styled.p`
  text-transform: uppercase;
  text-align: center;
  width: 100%;
  font-size: 1.5rem;
  white-space: nowrap;
  letter-spacing: 3px;
  margin: 0;
  animation-fill-mode: both;

  ${mobile} {
    display: flex;
    align-items: center;
    justify-content: center;
    animation-name: none !important;
    letter-spacing: 2px;
  }

  ${tablet} {
    font-size: 1.5rem;
  }

  ${phone} {
    font-size: 0.9rem;
  }

  ${phoneSm} {
    font-size: 0.65rem;
  }
`;

export const ResultUserChoiceLabel = styled(ResultChoiceLabel)`
  animation-name: ${moveLeft};
  grid-area: 1 / 1;

  ${mobile} {
    grid-area: 2 / 1;
  }
`;

export const ResultHouseChoiceLabel = styled(ResultChoiceLabel)`
  animation-name: ${moveRight};
  grid-area: 1 / 3;

  ${mobile} {
    grid-area: 2 / 2;
  }
`;

const ResultChoice = styled.div<{ readonly size: number }>`
  background: radial-gradient(
      circle,
      rgba(0 0 0 / 0.1) 60%,
      rgba(0 0 0 / 0) 60%
    )
    center no-repeat;
  animation-fill-mode: both;
  border-radius: 50%;

  ${mobile} {
    animation-name: none !important;
  }
`;

export const ResultUserChoice = styled(ResultChoice)`
  animation-name: ${moveLeft};
  grid-area: 2 / 1;

  ${mobile} {
    grid-area: 1 / 1;
  }
`;

export const ResultHouseChoice = styled(ResultChoice)`
  animation-name: ${moveRight};
  grid-area: 2 / 3;

  ${mobile} {
    grid-area: 1 / 2;
  }

  ${OptionButton} {
    animation-name: ${fadeIn};
    animation-fill-mode: both;
  }
`;

export const Result = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  animation-name: ${growAndfadeIn};
  animation-fill-mode: both;
  z-index: 1;
  grid-area: 2 / 2;
  width: 100%;

  ${mobile} {
    animation-name: ${fadeIn};
    grid-area: 3 / 1 / 3 / span 2;
  }
`;

export const ResultMessage = styled.p`
  text-transform: uppercase;
  white-space: nowrap;
  font-size: 4rem;
  font-weight: bold;
  margin: 0 0 15px 0;

  ${mobile} {
    margin: 0 0 10px 0;
  }

  ${tablet} {
    font-size: 4rem;
  }

  ${phone} {
    font-size: 3.5rem;
  }

  ${phoneSm} {
    font-size: 3rem;
  }
`;

export const RetryButton = styled(Button)`
  &:hover {
    color: hsl(349, 70%, 56%);
  }

  ${mobile} {
    width: 60%;
    max-width: 280px;
  }

  ${phoneSm} {
    font-size: 0.8rem;
  }
`;

export const ResultContainer = styled.div<ResultProps>`
  height: 100%;
  width: 100%;
  display: grid;
  grid-template: 0.6fr max-content / max-content 20% max-content;
  place-items: center;
  place-content: start center;
  gap: 0 10%;

  ${mobile} {
    grid-template: max-content max-content 0.75fr / 1fr 1fr;
    place-content: end;
    gap: 5%;
  }

  ${Result}, ${ResultChoice}, ${ResultChoiceLabel} {
    animation-duration: ${(props) => props.showResultDuration}s;
    animation-delay: ${(props) => props.showResultDelay}s;
  }

  ${ResultChoice} {
    ${OptionButton} {
      animation-duration: ${(props) => props.showHouseChoiceDuration}s;
      animation-delay: ${(props) => props.showHouseChoiceDelay}s;
    }
  }

  ${(props) => (props.userWins ? ResultUserChoice : ResultHouseChoice)} {
    position: relative;

    ${OptionButton} {
      position: relative;

      &:before {
        content: "";
        width: 275%;
        height: 275%;
        position: absolute;
        border-radius: 50%;
        top: -87.5%;
        left: -87.5%;
        z-index: -1;
        background-image: radial-gradient(
          circle,
          rgba(255, 255, 255, var(--result-rg-first)) 40%,
          rgba(255, 255, 255, var(--result-rg-first-border)) 40%,
          rgba(255, 255, 255, var(--result-rg-second)) 55%,
          rgba(255, 255, 255, var(--result-rg-second-border)) 55%,
          rgba(255, 255, 255, var(--result-rg-third)) 70%,
          rgba(255, 255, 255, var(--result-rg-third-border)) 70%,
          rgba(255, 255, 255, 0) 0%
        );
        animation: ${radialBackgroundEffect} 1.2s ease
          ${(props) => props.winnerBackgroundEffectDelay}s infinite alternate
          both;

        ${mobile} {
          width: 200%;
          height: 200%;
          top: -50%;
          left: -50%;
          background-image: radial-gradient(
            circle,
            rgba(255, 255, 255, var(--result-rg-first)) 43%,
            rgba(255, 255, 255, var(--result-rg-first-border)) 43%,
            rgba(255, 255, 255, var(--result-rg-second)) 57%,
            rgba(255, 255, 255, var(--result-rg-second-border)) 57%,
            rgba(255, 255, 255, var(--result-rg-third)) 100%,
            rgba(255, 255, 255, var(--result-rg-third-border)) 100%,
            rgba(255, 255, 255, 0) 0%
          );
        }
      }
    }
  }
`;
