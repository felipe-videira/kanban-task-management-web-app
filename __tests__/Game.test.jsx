import "jsdom-global/register";
import "jest-styled-components";

import React from "react";
import { render, mount } from "enzyme";
import { navigateMock, useParamsMock } from "../__mocks__/react-router-dom";
import Game from "../src/pages/Game";

const gameConfig = [
  {
    name: "original",
    rules: "src/images/images-rules.svg",
    listSetup: {
      pointingUp: false,
    },
    options: [
      {
        name: "paper",
        icon: "src/images/icon-paper.svg",
        color: ["hsl(230, 89%, 62%)", "hsl(230, 89%, 65%)"],
        winRules: {
          scissors: false,
          rock: true,
        },
      },
      {
        name: "scissors",
        icon: "src/images/icon-scissors.svg",
        color: ["hsl(39, 89%, 49%)", "hsl(40, 84%, 53%)"],
        winRules: {
          rock: false,
          paper: true,
        },
      },
      {
        name: "rock",
        icon: "src/images/icon-rock.svg",
        color: ["hsl(349, 71%, 52%)", "hsl(349, 70%, 56%)"],
        winRules: {
          scissors: true,
          paper: false,
        },
      },
    ],
  },
];
jest.mock("../src/services/score", () => ({
  __esModule: true,
  get: jest.fn(),
  save: jest.fn(),
  clear: jest.fn(),
  setSaveOnExitListener: jest.fn(),
}));
jest.mock("../src/utils/checkFile", () => ({
  __esModule: true,
  default: () => true,
}));
jest.mock("../src/utils/isColor", () => ({
  __esModule: true,
  default: () => true,
}));

jest.mock("../src/gameConfig.json", () => gameConfig);

// /utils/isMobile, window?

// snapshot
// user click
// size
// match logic
// score
describe("Game", () => {
  it("should render correctly", () => {
    useParamsMock.mockImplementation(() => {
      return { gameName: "original" };
    });

    const wrapper = render(<Game />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call "navigate" if the game does not exist', () => {
    useParamsMock.mockImplementation(() => {
      return { gameName: "wrong" };
    });

    mount(<Game />);

    expect(navigateMock).toHaveBeenCalledWith("/");
  });

  it("should ", () => {
    useParamsMock.mockImplementation(() => {
      return { gameName: "original" };
    });

    const wrapper = mount(<Game />);

    expect(wrapper.find("h1").text()).toBe(gameConfig[0].name);
  });
});
