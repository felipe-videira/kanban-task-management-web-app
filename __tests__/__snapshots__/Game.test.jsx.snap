// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  margin: 0 auto;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 100vh;
  padding: 0 5%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border: 3px solid #ffffff59;
  border-radius: 20px;
  padding: 2.5% 5%;
}

.c2 {
  text-transform: uppercase;
  width: 10%;
}

.c3 {
  height: 0px;
  width: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 100px;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 0;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    width: 75%;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h1
      class="c2"
    />
    <div
      class=""
    />
  </div>
  <div
    class="c3"
  />
</div>
`;

exports[`Game should render correctly with all props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  margin: 0 auto;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 100vh;
  padding: 0 5%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border: 3px solid #ffffff59;
  border-radius: 20px;
  padding: 2.5% 5%;
}

.c2 {
  text-transform: uppercase;
  width: 10%;
}

.c3 {
  height: 0px;
  width: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 100px;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 0;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    width: 75%;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h1
      class="c2"
    />
    <div
      class=""
    />
  </div>
  <div
    class="c3"
  />
</div>
`;
